apps/easypid/src/features/receive/FunkeOpenIdCredentialNotificationScreen.tsx

```diff
@@ -109,188 +62,40 @@ export function FunkeCredentialNotificationScreen() {
     [isDevelopmentModeEnabled]
   )
 
-  const shouldUsePinForPresentation = useShouldUsePinForSubmission(credentialsForRequest?.formattedSubmission)
-  const preAuthGrant =
-    resolvedCredentialOffer?.credentialOfferPayload.grants?.['urn:ietf:params:oauth:grant-type:pre-authorized_code']
-  const txCode = preAuthGrant?.tx_code
+  const shouldUsePinForPresentation = useShouldUsePinForSubmission(
+    resolvedOffer?.flow === 'auth-presentation-during-issuance'
+      ? resolvedOffer.credentialsForProofRequest.formattedSubmission
+      : undefined
+  )
 
   useEffect(() => {
-    resolveOpenId4VciOffer({
-      agent: paradym.agent,
-      offer: {
-        uri: params.uri,
-      },
-      authorization,
-    })
-      .then(({ resolvedAuthorizationRequest, resolvedCredentialOffer }) => {
-        setResolvedCredentialOffer(resolvedCredentialOffer)
-        setResolvedAuthorizationRequest(resolvedAuthorizationRequest)
+    if (resolvedOffer) return
+
+    paradym.openid4vc
+      .resolveCredentialOffer({
+        offer: params,
+        authorization,
+      })
+      .then((result) => {
+        setResolvedOffer(result)
```

Comment:
I could also have the `resolveOffer` object contain a function which does the `acquireCredential`. Might be a bit overkill, but we think that that makes it easier, I can change it instead of a paradym sdk call.